### ISSUE TITLE
Support the XDG Base Directory Specification

### ISSUE THREAD

**Issue by itaranto** [cgdb/cgdb#203]
Add support for the XDG Base Directory Specification
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Right now cgdb dumps logs and command history into **$HOME/.cgdb/** and also looks for the cgdbrc file in here as well.

It would be nice if the cgdbrc config file  could be located under **$XDG_CONFIG_HOME/cgdb/** (fallback to $HOME/.config/cgdb/) and logs and other volatile data under $XDG_DATA_HOME/cgdb/ (fallback to $HOME/.local/share/cgdb/).

Maybe a directory for configuration is not needed, if that is preferred only create **$XDG_CONFIG_HOME/cgdbrc**.

**Comment by yashlala** [comment]
An XDG style fallback seems easy; sourcing an environment variable for the data-folder location while defaulting to `.cgdb` if unset seems like a tidy little optional feature that won't break anything existing while letting users control their home directory. 

@itaranto are you planning on/already writing a pr, or should I take a stab at it?

**Comment by brasko** [comment]
Hi Itaranto,

I'm going to close this issue as yashlala has implemented the following feature in CGDB,
  * CGDB now supports a configurable home directory. If the environment
    variable 'CGDB_DIR' is set to a directory name, CGDB will use the
    specified directory instead of '~/.cgdb/'.

I believe you can use this to support the workflow you have mentioned above.

Please let us know otherwise.

**Comment by ideasman42** [comment]
@brasko could this be reconsidered?

Since 2019, `XDG_CONFIG_HOME` is by far the most widely supported an used location for config files these days.
Supported by GDB as well as other tools such as VIM/Emacs/TMUX... etc.
`cgdb` is one of the few programs I use that doesn't support `XDG_CONFIG_HOME`.

Even though it can be configured, it's inconvenient to have to do this for each program.

**Comment by yashlala** [comment]
I don't have a stake in this project, so I won't speak for @brasko. I'll just share my thoughts. 

I considered doing what you wrote above, but decided it'd be a more code for questionable benefit. 

_Why "a lot more code"?_: CGDB puts all of its files in `~/.cgdb`...regardless of whether they're config or data files. Modern applications will put config files in `$XDG_CONFIG_HOME` and everything else in `$XDG_DATA_HOME`. So you'd need to go split the files between the two locations...while checking for fallback locations. 

_Why "questionable benefit"?_: this is a pretty small legacy project, and the number of people interested in implementing this was even smaller. @brasko has made a great tool that's very useful at what it does...so I moved `~/.cgdb` without complicating either of our lives. 

> Even though it can be configured, it's inconvenient to have to do this for each program.

I decided to take this inconvenience. IMO if you see greater benefit, then feel free to send a PR. 
You might still be able to do it in a backwards compatible way (checking for `~/.cgdb` and `$CGDB_DIR`, among other things). **Just make sure to consider the config-data split I mentioned before!**
